src/AutoRest.CSharp/Common/Utilities/StringExtensions.cs

```diff
@@ -8,12 +8,18 @@
 using System.Text;
 using System.Text.RegularExpressions;
 using Humanizer;
+using Humanizer.Inflections;
 using Microsoft.CodeAnalysis.CSharp;
 
 namespace AutoRest.CSharp.Utilities
 {
     internal static class StringExtensions
     {
+        static StringExtensions()
+        {
+            Vocabularies.Default.AddUncountable("data");
```

Comment:
The library treats `datum` as the singular form for `data`. I think we want to use `data` in both singular and plural cases for .NET SDK, so customizing `add` as uncountable here.